### PR TITLE
Overhaul makefile tutorial

### PR DESCRIPTION
**Pull request by awdeorio**
Complete rewrite of the Makefile tutorial to make it more stand-alone.  It no longer relies on a working set of EECS 280 Project 1 starter files.

### CODE REVIEW

File: docs/make.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+Run the regression test automatically with `make`.
+```console
+$ make test
+./main.exe < main_test.in > main_test.out
+diff main_test.out main_test.out.correct
+```
+
+<div class="primer-spec-callout info" markdown="1">
+**Pro-tip:** Run commands in parallel with `make -j`.
```
Comment (ohjuny) [inline docs/make.md:131]: I didn't know this!

File: docs/make.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+A *target* is usually the name of a file that is created by the rule, like `main.exe`.
+
+A *prerequisite* is a file that is used as input to create the target, like `main.cpp`.  There could be multiple prerequisites.
+
+A *recipe* is a command that `make` runs, like `g++ main.cpp -o main.exe`.  There might be more than one command.
+
+<div class="primer-spec-callout warning" markdown="1">
+**Pitfall:** Every recipe line must start with a TAB character.
```
Comment (ohjuny) [inline docs/make.md:161]:
```suggestion
**Pitfall:** Every recipe line must start with a <kbd>TAB</kbd> character.
```

File: docs/make.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+This is a concrete example of a rule.
+```make
+main.exe: main.cpp
+	g++ main.cpp -o main.exe
+```
+
+A *target* is usually the name of a file that is created by the rule, like `main.exe`.
+
+A *prerequisite* is a file that is used as input to create the target, like `main.cpp`.  There could be multiple prerequisites.
```
Comment (ohjuny) [inline docs/make.md:156]: Perhaps a line that mentions that .cpp files should be used here, not .h files. This could be confusing at first.

**Review by ohjuny — COMMENTED** [review]
Added a couple of nit picks, generally LGTM.

I am wondering, though, what the rationale was behind removing the "touch" command from Basic Commands? In this tutorial we could create the Makefile and test IO files using touch.

**Review by ohjuny — APPROVED** [review]
Just realized I can approve while submitting feedback.

File: docs/make.md (hunk 1 above)
Comment (awdeorio) [reply docs/make.md:131]: It makes the tests on P2 and P5 in particular run MUCH faster!

File: docs/make.md (hunk 3 above)
Comment (awdeorio) [reply docs/make.md:156]: I don't think I understand.  Here's two different examples that are pretty common:

```make
stats.cpp : stats.h

main.exe : main.cpp
  g++ main.cpp -o main.exe
```

Comment (ohjuny) [reply docs/make.md:156]: Oh perhaps I don't understand Makefiles well enough...

**Comment by awdeorio** [comment]
> I am wondering, though, what the rationale was behind removing the "touch" command from Basic Commands? In this tutorial we could create the Makefile and test IO files using touch.

I removed the touch command because I don't think students will use it.  In the past, we used it in a project setup section to create a file.  Now, we have students add a new file using instructions from an IDE tutorial.

Please tell me if I'm missing something!

**Comment by jamesjuett** [comment]
LGTM! Decoupling from P1 makes a lot of sense.

**Review by jamesjuett — APPROVED** [review]

**Pull request merged**